consensus/polybft/consensus_runtime.go

```diff
@@ -705,20 +707,10 @@ func (c *consensusRuntime) deliverMessage(msg *TransportMessage) (bool, error) {
 // and ending at the last block of previous epoch
 func (c *consensusRuntime) calculateUptime(currentBlock *types.Header, epoch *epochMetadata) (*CommitEpoch, error) {
 	uptimeCounter := map[types.Address]uint64{}
+	blockHeader := currentBlock
+	epochID := epoch.Number
 
-	if c.config.PolyBFTConfig.EpochSize < (uptimeLookbackSize + 1) {
-		// this means that epoch size must at least be 3 blocks,
-		// since we are not calculating uptime for lastBlockInEpoch and lastBlockInEpoch-1
-		// they will be included in the uptime calculation of next epoch
-		return nil, errors.New("epoch size not large enough to calculate uptime")
-	}
-
-	calculateUptimeForBlock := func(header *types.Header, validators AccountSet) error {
-		blockExtra, err := GetIbftExtra(header.ExtraData)
-		if err != nil {
-			return err
-		}
-
+	calculateUptimeForBlock := func(blockExtra *Extra, validators AccountSet) error {
```

Comment:
I would move this closure to a function returning `uptimeCounter, err`, it will be more performant and simplify enclosing function.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/897/commits/7e3625bb48d731068c5c64aebc56aafe8d42ffa1